src-tauri/src/main.rs

```diff
@@ -161,6 +162,56 @@ async fn set_monero_address(
     Ok(())
 }
 
+#[tauri::command]
+async fn set_application_language(
+    state: tauri::State<'_, UniverseAppState>,
+    application_language: String,
+) -> Result<(), String> {
+    drop(
+        state
+            .config
+            .write()
+            .await
+            .set_application_language(application_language.clone())
+            .await,
+    );
+    Ok(())
+}
+
+#[tauri::command]
+async fn resolve_application_language(
+    state: tauri::State<'_, UniverseAppState>,
+) -> Result<String, String> {
+    if state
+        .config
+        .read()
+        .await
+        .has_system_language_been_proposed()
+    {
+        Ok(state.config.read().await.application_language().to_string())
+    } else {
+        let system_language = get_locale().unwrap_or_else(|| String::from("en-US"));
+        drop(
+            state
```

Comment:
Rather take one write lock and do both operations, otherwise we could get deadlocks